care/facility/api/serializers/patient_consultation.py

```diff
@@ -165,8 +184,46 @@ def update(self, instance, validated_data):
 
         consultation = super().update(instance, validated_data)
 
+        try:
+            health_details_data = self.context["request"].data[
+                "new_health_details"
+            ]
+            vaccination_history = health_details_data.pop(
+                "vaccination_history", []
+            )
+            PatientHealthDetails.objects.filter(
+                id=consultation.last_health_details.id
+            ).update(**self.context["request"].data["new_health_details"])
+            consultation.last_health_details = (
+                PatientHealthDetails.objects.get(
+                    id=consultation.last_health_details.id
+                )
+            )
+            vaccines = []
+            for vaccine in vaccination_history:
+                vaccines.append(
+                    Vaccine(
+                        health_details=consultation.last_health_details,
+                        **vaccine,
+                    )
+                )
+            if vaccines:
+                Vaccine.objects.bulk_create(vaccines, ignore_conflicts=True)
+            consultation.save(update_fields=["last_health_details"])
```

Comment:
the id does not change, right? so do we need to save the consultation again?

Reply:
@vigneshhari In the request, there is a `new_health_details` field through which we are creating a `health_details` record. After this, we have to display this record as the `last_health_details` field in the consultation API. So we are basically initializing the field with the record after the consultation object has been created. That's why I am saving the consultation object again (only doing a partial update)